proposed/http-client/http-client.md

```diff
@@ -73,17 +73,14 @@ use Psr\Http\Message\RequestInterface;
 use Psr\Http\Client\Exception;
 
 /**
- * Exception for when a request failed, providing access to the failed request.
- *
- * This could be due to an invalid request, example when you can seek in the request body stream
- * or due invalid request object. 
+ * Exception for when a request failed.
  */
 interface RequestException extends Exception
 {
   /**
    * Returns the request.
    *
-   * The request object MAY be the same as passed to HttpClient::sendRequest()
+   * The request object MAY NOT be the same as passed to HttpClient::sendRequest()
```

Comment:
Immutability comes from PSR-7. We should mention it where we state we use PSR-7.